build-support/functions/10-util.sh

```diff
@@ -667,3 +667,20 @@ function go_mod_assert {
    fi
    return 0
 }
+
+function get_consul_module_versions {
+  local module_directories
+  module_directories=( "." "api" "envoyextensions" "proto-public" "sdk" "troubleshoot")
+  for module_dir in "${module_directories[@]}"; do
+    echo "Module versions for directory: '$module_dir':"
+    echo "--------------"
+    (cd "$module_dir" && go list -m all | grep -e github.com/hashicorp/consul/api \
+    -e github.com/hashicorp/consul/envoyextensions \
+    -e github.com/hashicorp/consul/proto-public \
+    -e github.com/hashicorp/consul/sdk \
+    -e github.com/hashicorp/consul/troubleshoot)
```

Comment:
~ Super minor and optional: if you add a `grep -v “consul/$module_dir”` at the end, you could omit the self-report without a version. This would make the leaf node results blank but that seems fine if we add a note?

Reply:
great call.  corrected in 2ae80130911529a8d7c7258027d690a00f720198